src/common/Input.js

```diff
@@ -43,6 +43,8 @@ export default class Input extends PureComponent<Props> {
     this.textInput.clear();
   };
 
+  formatText = (text) => text.length > 35 ? text.substring(0, 35).concat('...') : text;
```

Comment:
how does this magic number `35` arrived? 🤔 
What if orientation is landscape, or on tablet which have larger screen size

Reply:
Sorry, I didn't think about those. What shall I do to handle those cases?

Reply:
You can measure the text's dimensions with some functions.
There are components that do that for React Native (they rely on native code)
This though I would categorize as 'too much effort' for that task.

Reply:
Consider Googling for this issue. Is it reported in React Native? Is it easy to fix if reported or is it something that is unlikely to get fixed ever?

Reply:
Since characters are different widths for different screens and Native Fonts we can't get an accurate answer.

The best solution I can think of is to get the device width and divide it by the presumed width of a single character. The width of the character is largely variant but is around `60%` of the font size.

`no_chars = deviceWidth / (0.6*font_size)`

Reply:
Of course, we can. There are native calls that allow you to measure precisely the width of a given text in a given font with a given font size.
The question really is, should we half-fix a minor issue that happens only infrequently?
I am not in favor of a half fix. Sometimes no fix is better than an improper fix, because at least you did not spend any effort on it, and it is still open to fixing later (or auto-fixing if React Native address this)

Reply:
@aswinmohanme https://youarenotsosmart.com/2011/03/25/the-sunk-cost-fallacy/

Reply:
@borisyankov mind-blowing article 🔥

Reply:
Hmm, @HarshulSharma000 I am not sure what you suggest is correct, but that leads me to another idea. You or @pulkonet can try to fix it this way...
* only in the MultilineInput.js file 
* multilne should be `true` if there is any text in the input
* it should be `false` if the input is empty

This way we will show the placeholder as a single line.
Not sure how it will look visually, maybe it will have a nice ellipsis automatically, or it will just hide off screen which I think is still satisfactory.

Reply:
@pulkonet @borisyankov I think I got the problem to some extent. Actually in ComposeBox.js the max and min  height is hardcoded as a result its child(Our multiline input here) is not able to resize as it wishes to. So I guess we need to increase the hardcoded intial value. If we just increase the number of lines within TextInput it ends up being cut off as specified by pulkonet. I think we should avoid parent hardcoded styling and let the parent grow as its children grow.

Reply:
> @pulkonet @borisyankov I think I got the problem to some extent. Actually in ComposeBox.js the max and min height is hardcoded as a result its child(Our multiline input here) is not able to resize as it wishes to. So I guess we need to increase the hardcoded intial value. If we just increase the number of lines within TextInput it ends up being cut off as specified by pulkonet. I think we should avoid parent hardcoded styling and let the parent grow as its children grow.

@HarshulSharma000 Actually we don't need it. I am working on #2039, and now we are no more dependent on any heights.
Just image a case when hint is of two line, then in this case we only want hint to be shown in 1 line with dots in the end. 
Summary: We only want one line when there is no text.

Reply:
> only in the MultilineInput.js file
multilne should be true if there is any text in the input
it should be false if the input is empty

@borisyankov I tried this one long time ago 😅, but the problem is when we toggle multiline prop, input losses focus. So focus & selection needs to be preserved while toggling. Which needs to done in the RN repo. I am almost done with the Android part, hoping to get this fixed soon :)

Reply:
@jainkuniya indeed, I tried this myself and saw it is not working as expected. They are creating a new component when you toggle the `multiline` prop.